### ISSUE TITLE
[Bug]: iOS crash on clone with 4.5k files

### ISSUE THREAD

**Issue by juanbretti** [denolehov/obsidian-git#475]
### Describe the bug

I am trying to clone my repository and it crashes.

### Relevant errors (if available)

```shell
Crash, and restarts the applications.
```

### Steps to reproduce

I am running:
- iOS 16.3.1
- Obsidian 1.4.3 (95) 
- API v1.1.16
- Obsidian Git 2.19.0

To minimize the problems. On Windows:
- I had deleted completely the repository 
- Started from scratch an Obsidian vault.
- I added to this new repository my 4.5k files.
- Installed Obsidian Git on Windows.
- Cloned this repository to GitHub.com
- On GitHub, the repository is OK.

On my iPhone 12 with 256 GB of storage:
- I installed Obsidian from the AppStore.
- Created a new vault.
- Enabled Community plugins
- Installed Obsidian Git.
- Enabled the plugin.
- Run the command "Obsidian Git: Clone an existing remote repo"
- Typed my url with the personal token
- Selected "Vault Root"
- The remote has .obsidian, so click on "YES"
- Click on "DELETE ALL YOUR LOCAL CONFIG AND PLUGINS"
- Depth: I tried with blank and 1

The Cloning popup shows, but never clones anything.
It does delete all my CONFIG and PLUGINGS. So, when the applications restarts, Obsidian Git is no longer available.

On mobile: Is there any trick I am missing to clone a repository with high number of files?

### Expected Behavior

I am running:
- iOS 16.3.1
- Obsidian 1.4.3 (95) 
- API v1.1.16
- Obsidian Git 2.19.0

To minimize the problems. I had deleted completely the repository and started from scratch an Obsidian vault.
I added to this new repository my 4.5k files.
Installed Obsidian Git on Windows.
Cloned this repository to GitHub.com

On my iPhone 12 with 256 GB of storage:
- I installed Obsidian from the AppStore.
- Created a new vault.
- Enabled Community plugins
- Installed Obsidian Git.
- Enabled the plugin.
- Run the command "Obsidian Git: Clone an existing remote repo"
- Typed my url with the personal token
- Selected "Vault Root"
- The remote has .obsidian, so click on "YES"
- Click on "DELETE ALL YOUR LOCAL CONFIG AND PLUGINS"
- Depth: I tried with blank and 1

The Cloning popup shows, but never clones anything.
It does delete all my CONFIG and PLUGINGS. So, when the applications restarts, Obsidian Git is no longer available.

### Addition context

Being able to clone my personal repository on mobile, as the same way I am able to clone on desktop (I use Windows).

### Operating system

iOS

### Installation Method

Other

### Plugin version

Obsidian Git 2.19.0

### References

I read the other **issues**:
- https://github.com/denolehov/obsidian-git/issues/307
- https://github.com/denolehov/obsidian-git/issues/435
- https://github.com/denolehov/obsidian-git/issues/338
- https://github.com/denolehov/obsidian-git/issues/384

**Comment by Vinzent03** [comment]
Sorry, but your vault is simply too big for the underling library to handle on a mobile device. I see nothing I can do here.

**Comment by juanbretti** [comment]
Thank you @Vinzent03 for your reply. 

Is there any way to have a different `.gitignore` on my mobile?
Because to be honest, I don't need all the `.md` on my mobile. But I don't want to have two different **vaults**.

Thanks!

**Comment by Vinzent03** [comment]
No it is not, because just a different gitignore won't help, because once it's committed a gitignore doesn't change anything. You would need two different repositories.

**Comment by juanbretti** [comment]
If I may take advantage of your knowledge...
Do you think [Git Submodules](https://git-scm.com/book/en/v2/Git-Tools-Submodules) could solve my case?

**Comment by Vinzent03** [comment]
Haven't worked with subtree yet, but submodule could work. You would need to use the mobile only part as a submodule on desktop.

**Comment by Vinzent03** [comment]
No that's still correct. A submodule is just a normal git repository. The mobile only part is a repository and the outer desktop part is another repository, which contains the mobile part as submodule. So submodule is only needed on desktop.

**Comment by juanbretti** [comment]
Thank you @Vinzent03 .
Here are my steps to solve my problem:

- Create a new repository to host my mobile-light-small number of .md
- On my PC:

```shell
$ git submodule add https://github.com/xxx/obsidian-main.git Main
$ git add .
$ git commit -m "add"
$ git push
$ git pull --recurse-submodules
$ git config submodule.recurse true
```

- Enable at Obsidian Git "Update submodules"
- Use the new https://github.com/xxx/obsidian-main.git to connect Obsidian to my iPhone.
- Bingo!

**Comment by juanbretti** [comment]
Pufff, I was able to create the submodule... Synchronize the **submodule** on my PC.
... But on my iPhone, still **crash**.
And now the repository has 107 `.md` files.

Any ideas?

What I am doing:
- I have a clean Vault.
- Enable "Community plugins"
- Install "Obsidian Git"
- "Obsdian Git: Clone an existing remote repo"
- Using my repository URL as: `https://passs@github.com/xxxx/obsidian-main.git`
- Vault Root
- NO (my small repository does not have `.obsidian`)
- 1 or empty: tried both options

Then I get:
- Cloning new repo on "."
- another quick banner
- Crash, and nothing happens

Any ideas?

**Comment by Vinzent03** [comment]
I'm really sorry but I don't think I can help you there. I guess the repo is still too big for your device

**Comment by juanbretti** [comment]
Just to complete the situation.
I was able to make it work, by reducing the size (in number of `.md` and MiB) of the remote repository that is syncing on my iPhone.

Another learning, just to be complete... In case of failure...
To remove a **submodule**, follow https://gist.github.com/myusuf3/7f645819ded92bda6677
- Remove the submodule entry from .git/config
	`git submodule deinit -f path/to/submodule`
- Remove the submodule directory from the superproject's .git/modules directory (adapt the code to Windows if necessary)
	`rm -rf .git/modules/path/to/submodule`
- Remove the entry in `.gitmodules` and remove the submodule directory located at path/to/submodule
	`git rm -f path/to/submodule`
- Also delete `.gitmodules`

**Comment by andrewboldi** [comment]
For others hitting this on iOS: I spent a long session debugging the same crash and found the root cause is upstream in isomorphic-git, not the plugin — packfiles get loaded into the WebView heap in full during checkout, which iOS kills silently once the vault is non-trivial in size. It's the memory model documented in #572, not something fixable in obsidian-git's wrapper without the long-blocked wasm-git migration.

I built a separate plugin that works around it by skipping the git wire protocol entirely on the client and using GitHub's REST + Git Data APIs instead — one file at a time, bounded memory per file, **real git commits** on the GitHub side: **[Vault Sync (REST)](https://github.com/andrewboldi/obsidian-vault-sync)**.

Tested in production on iOS 26.3 with a ~840 file / 130 MB vault including images and PDFs. Two-way auto-sync at 1-minute intervals. Honest tradeoffs are in the README (GitHub-only; no client-side git history beyond a SHA pointer; no merge engine).

**Install via [BRAT](https://github.com/TfTHacker/obsidian42-brat) → Add Beta Plugin →** `https://github.com/andrewboldi/obsidian-vault-sync`

Will be in the community plugins directory once review completes (obsidianmd/obsidian-releases#12313). Big thanks to @Vinzent03 — debugging this in the open source the plugin made the root cause findable.
